crates/fsrt/src/main.rs

```diff
@@ -184,33 +211,103 @@ fn collect_sourcefiles<P: AsRef<Path>>(root: P) -> impl Iterator<Item = PathBuf>
         })
 }
 
-#[tracing::instrument(level = "debug")]
-fn scan_directory(dir: PathBuf, function: Option<&str>, opts: &Args) -> Result<()> {
-    let mut manifest_file = dir.clone();
-    manifest_file.push("manifest.yaml");
-    if !manifest_file.exists() {
-        manifest_file.set_extension("yml");
-    }
-    debug!(?manifest_file);
+pub fn scan_directory_test(forge_test_proj: Vec<MockForgeProject>) -> Option<Vec<Vulnerability>> {
+    scan_directory(PathBuf::new(), &Args::parse(), true, forge_test_proj)
+}
 
-    let manifest = fs::read_to_string(&manifest_file).into_diagnostic()?;
-    let manifest: ForgeManifest<'_> = serde_yaml::from_str(&manifest).into_diagnostic()?;
-    let name = manifest.app.name.unwrap_or_default();
+#[tracing::instrument(level = "debug")]
+fn scan_directory<'a>(
+    dir: PathBuf,
+    opts: &Args,
+    test: bool,
```

Comment:
Instead of having the `scan_directory` function have a dummy project that it uses if it's being run as a test, it might be cleaner to isolate the IO/FS calls that occur within the function and create a trait that encapsulates that behavior. You can also wrap all the swc parser and loader structs within a common struct that will be embedded within the types that actually implement the trait.

Reply:
I need some clarification here 😅. I am slightly confused, what would implement this trait.  Thank you in advance :) . sorry for the dumb question

Reply:
Fixed this, but lmk what you think. Thank you so much for the help again.